src/app/(routes)/page.tsx

```diff
@@ -1,40 +1,53 @@
 "use client";
 
+import { Modal } from "@/components/modals/modal";
 import { Card } from "@/components/ui/card";
 import KycApproval from "@/components/ui/kycApproval";
+import ValidatePassportScore from "@/components/validatePassportScore";
+import useIsClient from "@/lib/hooks/useIsClient";
 import { ConnectButton } from "@rainbow-me/rainbowkit";
-import { useRouter } from "next/router";
-import { useEffect, useState } from "react";
-import { useSignTypedData, useAccount } from "wagmi";
+import { useSearchParams } from "next/navigation";
+import { useEffect, useMemo, useState } from "react";
+import { useAccount } from "wagmi";
 
 export default function Home() {
-  const { isConnected } = useAccount();
-  const [kycParams, setKycParams] = useState({ clientId: "", repoName: "", repoOwner: "" });
-  const router = useRouter();
-  const { query } = router;
+  const [modalMessage, setModalMessage] = useState<string | null>(null);
+  const isClient = useIsClient();
+  const [score, setScore] = useState(0);
+  const [error, setError] = useState(false);
+  const { isConnected, address, chain } = useAccount();
+  const searchParams = useSearchParams();
+  const clientId = searchParams.get("clientId");
+  const repoName = searchParams.get("repoName");
+  const repoOwner = searchParams.get("repoOwner");
 
-  useEffect(() => {
-    if (query.clientId && query.repoName && query.repoOwner) {
-      setKycParams({
-        clientId: Array.isArray(query.clientId) ? query.clientId[0] : query.clientId,
-        repoName: Array.isArray(query.repoName) ? query.repoName[0] : query.repoName,
-        repoOwner: Array.isArray(query.repoOwner) ? query.repoOwner[0] : query.repoOwner
-      });
+  const kycParams = useMemo(() => {
+    if (clientId && repoName && repoOwner) {
+      return {
+        clientId,
+        repoName,
+        repoOwner
+      };
     }
-  }, [query.clientId, query.repoName, query.repoOwner]);
+    return { clientId: "", repoName: "", repoOwner: "" };
+  }, [clientId, repoName, repoOwner]);
```

Comment:
This seems to only take 3 variables and put them together in an object. Why do it at all and why use `useMemo` for this?